### ISSUE TITLE
distraction free mode

### ISSUE THREAD

**Issue by fedelibre** [wbsoft/frescobaldi#864]
This is a wish (please add the label), an idea that I find interesting.

I've just found out a nice feature of Gedit editor: if you click F11, it goes in [fullscreen mode](https://help.gnome.org/users/gedit/stable/gedit-full-screen.html.en). This is actually more than fullscreen: I would say that it's very close to what other editors call "distraction free mode". Here's what happens:

- top OS bar, header bar and tabs bar disappear so you have much more vertical space
- you see only the active file and forget the rest of open tabs: this helps to focus on what you are writing

Currently Frescobaldi has already a fullscreen mode which hides the top bar and the title bar.
What I'd like to see in Frescobaldi if a distraction free mode was implemented:

- hide also the menu bar and the toolbar (this may be shown when the mouse hover on the top, like it happens in gedit and [focuswriter](https://github.com/gottcode/focuswriter) also).
- hide the other open tabs
- hide all the panes except the logs and preview panes (a preference for this might be useful)

To sum up, it should go in fullscreen and show only the active tab: text editor and log pane on left, preview on right. Compilation will have to be triggered with the keyboard (but this is what many people already do, I guess).

What do you think about it?

**Comment by uliska** [comment]
I like this idea. Some comments:

Other than with regular editors a LilyPond score may be distributed to lots of files. That means it may be necessary to switch tabs while working on a single score. When having no tabs visible this can conveniently realized through point-and-click:

* The editor window always shows *one* file only, Clicking on a note may bring another tab to the front and hide the others.
* In that context compiling the score must make sure that all open (hidden) and changed files are saved.

However, the user should get a chance to manually see the list of open files to switch to another one. This should be done by some kind of popup/popover effect - but should *not* cause the upper edge of the editor window to change (when we would simply switch the tabbing area to visible there would be an annoying flickering movement of the document area).

---

I would suggest to even partly hide the log panel because *while writing* one doesn't want to see that. Maybe it should open while compiling and hide automatically, say, two seconds after the job is completed. It should then have - of course - an option *not* to do that as well as an unintrusive way to reopen the log (e.g. to inspect warnings and errors). I could imagine a small triangle somewhere at to lower edge of the document area for that.

Actually this might be an interesting option for the general mode right now. I'm usually annoyed by always having the log panel use up space.

---

A question is to what extent such a "Zen mode" has to be made user configurable. It would of course be much more convenient to simply say that this consists of two columns with file editor and music preview.

**Comment by wbsoft** [comment]
I like this idea of a "zen mode". I also think the default user interface should already be simple and as distraction free as possible.

**Comment by uliska** [comment]
I think the default user interface is already as simple as possible, with the user having control over which panels are open. The only thing that I can imagine to be an improvement separately from a Zen mode is the auto-hiding of the log as wished for in #869.
